### PR TITLE
Update storage-class.md

### PR DESCRIPTION
**Pull request by manavellamnimble**
@marccampbell Updates on default storage class analyzer

### CODE REVIEW

File: docs/source/analyze/storage-class.md [hunk 1]
```diff
@@ -3,6 +3,17 @@ title: Storage Class
 description: Analyzer to check for the presence of a storage class in the cluster
 ---
 
+### Use Cases
+
```
Comment (markpundsack) [inline docs/source/analyze/storage-class.md:7]: > The ability to check for a default storage class was introduced in Kots 1.19.0 and Troubleshoot 0.9.42.

File: docs/source/analyze/storage-class.md [hunk 2]
```diff
@@ -3,6 +3,17 @@ title: Storage Class
...
+There are two use cases for the Storage Class Analyzer:
+
+- Check for the presence of a specific storage class, in which case ```storageClassName``` must be provided (Example 1)
+- Check if there is a Storage Class set as default. The analyzer checks if there is any storage with the ```isDefaultStorageClass``` field set to true. (Examples 2 and 3)
+
+In the second case, all arguments are optional. If none are provided, default messages will indicate the user that no default Storage Class was found.
```
Comment (markpundsack) [inline docs/source/analyze/storage-class.md:13]:
```suggestion
In the second case, all arguments are optional. If none are provided, default messages will indicate that no default Storage Class was found.
```

File: docs/source/analyze/storage-class.md [hunk 3]
```diff
@@ -20,4 +31,38 @@ spec:
...
+kind: Preflight
+metadata:
+  name: preflight-sample
+spec:
+  analyzers:
+    - storageClass:
+        checkName: Check for default storage class
+        outcomes:
+          - fail:
+              message: No Storage Class set as default found
```
Comment (markpundsack) [inline docs/source/analyze/storage-class.md:47]:
```suggestion
              message: No default storage class found
```

File: docs/source/analyze/storage-class.md [hunk 4]
```diff
@@ -20,4 +31,38 @@ spec:
...
+          - pass:
+              message: Default storage class found
+```
+#### Example 3: Check for the presence of a default storage class using default messages and checkName
+
+Defaults for storageClass analyzer are:
+  - ```checkName``` = 'Default Storage Class'
+  - Fail Message = 'No Default Storage Class found'
```
Comment (salahalsaleh) [inline docs/source/analyze/storage-class.md:55]:
```suggestion
  - Fail Message = 'No default storage class found'
```

File: docs/source/analyze/storage-class.md [hunk 5]
```diff
@@ -3,6 +3,17 @@ title: Storage Class
...
+### Use Cases
+
+There are two use cases for the Storage Class Analyzer:
+
+- Check for the presence of a specific storage class, in which case ```storageClassName``` must be provided (Example 1)
+- Check if there is a Storage Class set as default. The analyzer checks if there is any storage with the ```isDefaultStorageClass``` field set to true. (Examples 2 and 3)
+
+In the second case, all arguments are optional. If none are provided, default messages will indicate the user that no default Storage Class was found.
+
+#### Example 1: Check for a specific Storage Class
```
Comment (markpundsack) [inline docs/source/analyze/storage-class.md:15]:
```suggestion
#### Example 1: Check for a specific storage class
```

File: docs/source/analyze/storage-class.md [hunk 6]
```diff
@@ -3,6 +3,17 @@ title: Storage Class
 description: Analyzer to check for the presence of a storage class in the cluster
 ---
 
+### Use Cases
+
+There are two use cases for the Storage Class Analyzer:
+
+- Check for the presence of a specific storage class, in which case ```storageClassName``` must be provided (Example 1)
+- Check if there is a Storage Class set as default. The analyzer checks if there is any storage with the ```isDefaultStorageClass``` field set to true. (Examples 2 and 3)
```
Comment (markpundsack) [inline docs/source/analyze/storage-class.md:11]:
```suggestion
- Check if there is a storage class set as default. The analyzer checks if there is any storage with the ```isDefaultStorageClass``` field set to true. (Examples 2 and 3)
```

**Pull request merged**